src/Chapter3.hs

```diff
@@ -970,6 +1086,27 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data Weekday
+  = Saturday
+  | Sunday
+  | Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  deriving (Show, Read, Eq, Ord, Bounded, Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend Saturday  = True
+isWeekend Sunday    = True
+isWeekend _         = False
+
+nextDay :: Weekday -> Weekday
+nextDay Friday  = Saturday
```

Comment:
That is right!
You could also use the instance of `Bounded` in here:

```suggestion
    | day == maxBound = minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂

Reply:
Thank you for suggestion. I read the document on minBound and maxBound and changed the code to use them.